### ISSUE TITLE
Basic 2D convolutional neural network

### ISSUE THREAD

**Issue by substructural** [substructural/deep-random-fields#2]
Implementation of a basic 2D convolutional network, with:
- fixed learning rate
- no regularisation
- leaky ReLU and softmax activation functions
- cross entropy error function
- optimisation with Stochastic Gradient Descent with minibatch
